pkg/schemadsl/compiler/compiler_test.go

```diff
@@ -562,27 +563,173 @@ func TestCompile(t *testing.T) {
+		{
+			"caveat missing parameter",
+			&someTenant,
+			`caveat foo(someParam int) {
+				anotherParam == 42
+			}`,
+			`ERROR: sometenant/foo:2:5: undeclared reference to 'anotherParam'`,
+			[]SchemaDefinition{},
+		},
+		{
+			"caveat missing parameter on a different line",
+			&someTenant,
+			`caveat foo(someParam int) {
+				someParam == 42 &&
+					anotherParam
+			}`,
+			`ERROR: sometenant/foo:3:6: undeclared reference to 'anotherParam'`,
+			[]SchemaDefinition{},
+		},
+		{
+			"caveat invalid expression type",
+			&someTenant,
+			`caveat foo(someParam int) {
+				someParam
+			}`,
+			`caveat expression must result in a boolean value`,
+			[]SchemaDefinition{},
+		},
+		{
+			"caveat invalid expression",
+			&someTenant,
+			`caveat foo(someParam int) {
+				someParam:{}
+			}`,
+			`ERROR: sometenant/foo:2:14: Syntax error: mismatched input ':'`,
+			[]SchemaDefinition{},
+		},
+		{
+			"caveat valid",
+			&someTenant,
+			`caveat foo(someParam int) {
+				someParam == 42
+			}`,
+			``,
+			[]SchemaDefinition{
+				namespace.MustCaveatDefinition(caveats.MustEnvForVariables(
+					map[string]caveattypes.VariableType{
+						"someParam": caveattypes.IntType,
+					},
+				), "sometenant/foo", "someParam == 42"),
+			},
+		},
+		{
+			"long caveat valid",
+			&someTenant,
+			`caveat foo(someParam int, anotherParam string, thirdParam list<int>) {
+				someParam == 42 && someParam != 43 && someParam < 12 &&
+				someParam > 56 && anotherParam == "hi there" && 42 in thirdParam
+			}`,
+			``,
+			[]SchemaDefinition{
+				namespace.MustCaveatDefinition(caveats.MustEnvForVariables(
+					map[string]caveattypes.VariableType{
+						"someParam":    caveattypes.IntType,
+						"anotherParam": caveattypes.StringType,
+						"thirdParam":   caveattypes.ListType(caveattypes.IntType),
+					},
+				), "sometenant/foo",
+					`someParam == 42 && someParam != 43 && someParam < 12 && someParam > 56 
+					&& anotherParam == "hi there" && 42 in thirdParam`),
+			},
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
 			require := require.New(t)
-			defs, err := Compile([]InputSchema{
+			compiled, err := Compile([]InputSchema{
 				{input.Source(test.name), test.input},
 			}, test.implicitTenant)
 
 			if test.expectedError != "" {
 				require.Error(err)
-				require.Equal(test.expectedError, err.Error())
+				require.Contains(err.Error(), test.expectedError)
 			} else {
 				require.Nil(err)
-				require.Equal(len(test.expectedProto), len(defs))
-				for index, def := range defs {
+				require.Equal(len(test.expectedProto), len(compiled.OrderedDefinitions))
+				for index, def := range compiled.OrderedDefinitions {
 					filterSourcePositions(def.ProtoReflect())
-					require.True(proto.Equal(test.expectedProto[index], def), "proto mismatch. expected: %v, found: %v", test.expectedProto[index], def)
+					expectedDef := test.expectedProto[index]
+
+					if caveatDef, ok := def.(*core.CaveatDefinition); ok {
```

Comment:
Not necessary with `proto.Equal` that we use; I only did this because the caveats produced *aren't* the exact same protos, because CEL encodes extra info into expression protos that makes them differ slightly from run to run